xsalsa20.js

```diff
@@ -53,7 +53,7 @@ function loadWebAssembly (opts) {
 
 function toUint8Array (s) {
   if (typeof atob === 'function') return new Uint8Array(atob(s).split('').map(charCodeAt))
-  return new (require('buf' + 'fer').Buffer)(s, 'base64')
+  return new (require('buf' + 'fer').Buffer).from(s, 'base64')
```

Comment:
you need to remove the new also still right?

Reply:
You're absolutely right. Weirdly enough it still works though 🤣:

```
> Buffer.from("abc")
<Buffer 61 62 63>
> new Buffer.from("abc")
<Buffer 61 62 63>
> (new Buffer).from("abc")
Uncaught:
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined
    at Function.from (buffer.js:327:9)
    at new Buffer (buffer.js:283:17)
    at repl:1:2
    at Script.runInThisContext (vm.js:120:20)
    at REPLServer.defaultEval (repl.js:431:29)
    at bound (domain.js:426:14)
    at REPLServer.runBound [as eval] (domain.js:439:12)
    at REPLServer.onLine (repl.js:759:10)
    at REPLServer.emit (events.js:322:22)
    at REPLServer.EventEmitter.emit (domain.js:482:12) {
  code: 'ERR_INVALID_ARG_TYPE'
}
> (node:4564) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability 
issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.   
> new (Buffer.from("abc"))
Uncaught TypeError: Buffer.from(...) is not a constructor
```

Do you want to leave `require('buf' + 'fer')` in? `Buffer` is global anyway.

Reply:
The buf + fer is a hack to avoid browserify bundling it in

Reply:
I see. I think most modern bundlers skip built-in node modules automatically, or at least provide options to mark them as external. Plus removing the require would also solve this issue. But of course that's another topic 👍